tools/build/libwebrtc/mrwebrtc.ps1

```diff
@@ -0,0 +1,270 @@
+    Write-Host $message
+}
+
+# Write a key-value pair with higlight on the value.
+function Write-KeyValue([string]$key, [string]$value)
+{
+    Write-Host -NoNewLine $key
+    Write-Host -Foreground Blue $value
+}
+
+# Initialize the environment variables
+function Initialize-BuildEnvironment
+{
+    Write-TaskStart "Setting up environment..."
+
+    $repoRoot = Get-RootFolder
+    Write-KeyValue "Repository root: " $repoRoot
+
+    # Add depot_tools folder to PATH
+    $externalFolder = Join-Path $repoRoot "external/"
+    $depotToolsPath = Join-Path $externalFolder "depot_tools"
+    if (!($env:PATH.StartsWith($depotToolsPath)))
+    {
+        # Prepend depot_tools path as required by Chromium
+        $env:PATH = "$depotToolsPath;$env:PATH"
+        Write-Host "Updated PATH to include depot_tools location"
+    }
+    else
+    {
+        Write-Host "PATH already includes depot_tools location"
+    }
+
+    Write-Host  "Use local Windows toolchain (DEPOT_TOOLS_WIN_TOOLCHAIN = 0)"
+    $env:DEPOT_TOOLS_WIN_TOOLCHAIN = "0"
+
+    Write-Host "Use Visual Studio 2019 (GYP_MSVS_VERSION = 2019)"
+    $env:GYP_MSVS_VERSION = "2019"
+}
+
+# Install the Google repository of libwebrtc into external/libwebrtc
+function Install-GoogleRepository
+{
+    $externalFolder = Get-ExternalFolder
+
+    # Prepare the folder
+    $repoFolder = Join-Path $externalFolder "libwebrtc"
+    New-Item $repoFolder -ItemType Directory -Force | Out-Null
+    $repoFolder = Resolve-Path $repoFolder
+    Write-Host "Installing Google's libwebrtc in $repoFolder"
+
+    Push-Location $repoFolder
+
+    # Download
+    Write-TaskStart "Fetching Google repository, this might take some time..."
+    try {
+        Invoke-Expression "fetch --nohooks --no-history webrtc"
+    }
+    catch {
+        Write-Failure "Failed to fetch Google repository 'webrtc'."
+        throw
+    }
+    
+    # Checkout + sync
+    $libwebrtcFolder = Join-Path $repoFolder "src" -Resolve
+    Set-Location $libwebrtcFolder
+    try {
+        Write-TaskStart "Checking out M80 branch, this might take some time..."
+        Invoke-Expression "git checkout branch-heads/3987"
+        Write-TaskStart "Synchronizing dependencies, this might take some time..."
+        Invoke-Expression "gclient sync"
+    }
+    catch {
+        Write-Failure "Failed to checkout M80 branch 'branch-heads/3987'."
+        throw
+    }
+
+    # Apply patches
+    $env:WEBRTCM80_ROOT = $libwebrtcFolder
+    Write-TaskStart "Patching M80 for UWP..."
+    Push-Location "../../winrtc/patches_for_WebRTC_org/m80"
+    try {
+        & .\patchWebRTCM80.cmd
+    }
+    catch {
+        Write-Failure "Failed to apply UWP patches on M80 branch."
+        throw
+    }
+    Pop-Location
+
+    Pop-Location
+}
+
+# Install dependencies
+function Install-Dependencies
+{
+    Write-TaskStart "Installing dependencies..."
+
+    # Check python
+    try
+    {
+        Get-Command "python" -ErrorAction Stop | Out-Null
+    }
+    catch
+    {
```

Comment:
nit: are we doing Egyptian braces? There are mixed styles.

Reply:
I don't quite know what's the canonical way in PowerShell. Seems like VSCode "Format Document" command produces some inline braces instead, so happy to change to that.

Reply:
It needn't block this PR. Just something for the future. BTW I went looking for opinions and hadn't realized that 
```
foo { 
  bar
}
```
and
```
foo
{
  bar
}

```
are quite different things to powershell. Which is why some people recommend K&R braces.
https://powershell.org/2013/07/powershell-great-debate-formatting-constructs/#comment-4968